frame/nomination-pools/src/lib.rs

```diff
@@ -1613,9 +1635,9 @@ pub mod pallet {
 			let current_era = T::StakingInterface::current_era();
 
 			let bonded_pool = BondedPool::<T>::get(member.pool_id)
-				.defensive_ok_or_else(|| Error::<T>::PoolNotFound)?;
+				.defensive_ok_or_else(|| -> Error<T> { DefensiveError::PoolNotFound.into() })?;
```

Comment:
You can shorten these. Same with the other defensive calls:  
```suggestion
				.defensive_ok_or::<Error<T>>(DefensiveError::PoolNotFound.into())?;
```
Not sure what @kianenigma preferred.

Reply:
@ggwpez The ```defensive_*``` functions take two generics, so I will stick with the first approach if there isn't a reason against it.

Reply:
Yea the second one can be inferred with `_` but whatever.

Reply:
There actually is a cost for creating a closure every time a defensive error is encountered, and in this situation, the cost of it is higher than simply creating a `DefensiveError`. Is there a good reason other than saving typing to use a closure here?

Reply:
@KiChjang I have fixed this issue, could you review now?